client/checkout/api/index.js

```diff
@@ -209,9 +227,36 @@ export default class WCPayAPI {
 			orderId = orderIdPartials[ 0 ];
 		}
 
+		// For platform checkout we need the capability to switch up the account ID specifically for
+		// the intent confirmation step, that's why we have a new function for this here.
+		const getStripe = () => {
+			const { locale, publishableKey } = this.options;
+			const accountIdForIntentConfirmation = getConfig(
+				'accountIdForIntentConfirmation'
+			);
+
+			// If this is a setup intent we're not processing a platform checkout payment so we can
+			// use the regular getStripe function.
+			if ( isSetupIntent ) {
+				return this.getStripe();
+			}
+
+			if ( accountIdForIntentConfirmation ) {
+				return this.createStripe(
+					publishableKey,
+					locale,
+					accountIdForIntentConfirmation
+				);
+			}
+
+			// When not dealing with a setup intent or platform checkout we need to force an account
+			// specific request in Stripe.
+			return this.getStripe( true );
+		};
+
 		const confirmAction = isSetupIntent
-			? this.getStripe().confirmCardSetup( clientSecret )
-			: this.getStripe( true ).confirmCardPayment( clientSecret );
+			? getStripe().confirmCardSetup( clientSecret )
+			: getStripe().confirmCardPayment( clientSecret );
```

Comment:
This part is kinda tricky IMO, and even more now with the extra condition for the different account ID. I wonder if we can make it easier to understand what's going on.

Some ideas came to mind:

- Perhaps we should call the `getStripe` function here something else, to help differentiate between this and `WCPayAPI.getStripe`?
- Or, does it make sense to create a function to return the intent confirmation (confirmCardSetup/confirmCardPayment)? We can at least dismiss the extra `isSetupIntent` check.

For example:

```JavaScript
const confirmIntention = () => {
	const { locale, publishableKey } = this.options;
	const accountIdForIntentConfirmation = getConfig(
		'accountIdForIntentConfirmation'
	);

	// If this is a setup intent we're not processing a platform checkout payment so we can
	// use the regular getStripe function.
	if ( isSetupIntent ) {
		return this.getStripe().confirmCardSetup( clientSecret );
	}

	if ( accountIdForIntentConfirmation ) {
		return this.createStripe(
			publishableKey,
			locale,
			accountIdForIntentConfirmation
		).confirmCardPayment( clientSecret );
	}

	// When not dealing with a setup intent or platform checkout we need to force an account
	// specific request in Stripe.
	return this.getStripe( true ).confirmCardPayment( clientSecret );
};

const request = confirmIntention()
```

- Or even, avoid this altogether by overriding `options.accountId` when initializing the WCPay API with `getConfig( 'accountIdForIntentConfirmation' )`.

```JavaScript
// Or could this be overwritten from the back-end instead of adding "accountIdForIntentConfirmation"?
const accountId = getConfig( 'accountIdForIntentConfirmation' ) ? getConfig( 'accountIdForIntentConfirmation' ) : getConfig( 'accountId' );

new WCPayAPI(
	{
		publishableKey,
		accountId,
		forceNetworkSavedCards: getConfig( 'forceNetworkSavedCards' ),
		locale: getConfig( 'locale' ),
	}
```

And we can keep this part of the code as it was before.

```JavaScript
const confirmAction = isSetupIntent
	? this.getStripe().confirmCardSetup( clientSecret )
	: this.getStripe( true ).confirmCardPayment( clientSecret );
```

PS: Haven't tested the code here though 😅.

Reply:
> Perhaps we should call the `getStripe` function here something else, to help differentiate between this and `WCPayAPI.getStripe?`

I did call them the same thing because they sort of serve the same purpose, but I can definitely see how that could be confusing 😅 

> avoid this altogether by overriding `options.accountId` when initializing the WCPay API with `getConfig( 'accountIdForIntentConfirmation' )`.

Unfortunately this won't work since the `WCPayAPI.getStripe()` function "caches" the Stripe SDK instance it creates, so we'll have to recreate the Stripe SDK instance here anyway. The reason being that the SDK should normally be instantiated with the store account ID, but in some cases (e.g. platform checkout intent confirmation) we need a different account ID, so we'll need 2 Stripe JS SDKs initialized at this point.

> does it make sense to create a function to return the intent confirmation (confirmCardSetup/confirmCardPayment)?

Hmm, I like this idea 🤔 

Let's go with that and see if that improves things 👏

Reply:
I tried this out in https://github.com/Automattic/woocommerce-payments/pull/3811/commits/c245507af26b1e264a92da94bd5f4e2e74981a5c, but not sure if it looks better 🤔 
What do you think?